lib/watchman.rb

```diff
@@ -12,35 +13,39 @@ class << self
     attr_accessor :port
     attr_accessor :test_mode
 
-    def submit(name, value, type = :gauge)
-      metric = metric_name_with_prefix(name)
-
-      case type
-      when :gauge  then statsd_client.gauge(metric, value)
-      when :timing then statsd_client.timing(metric, value)
-      when :count  then statsd_client.count(metric, value)
-      else raise SubmitTypeError.new("Submit type '#{type}' is not recognized")
-      end
-    end
-
-    def benchmark(name)
+    def benchmark(name, options = {})
       result = nil
 
       time = Benchmark.measure do
         result = yield
       end
 
-      submit(name, (time.real * 1000).floor, :timing)
+      timing(name, (time.real * 1000).floor, options)
 
       result
     end
 
-    def increment(name)
-      submit(name, 1, :count)
+    def timing(name, value, options = {})
+      submit(name, value, :timing, options)
+    end
+
+    def increment(name, options = {})
+      submit(name, 1, :count, options)
+    end
+
+    def decrement(name, options = {})
+      submit(name, -1, :count, options)
     end
 
-    def decrement(name)
-      submit(name, -1, :count)
+    def submit(name, value, type = :gauge, options = {})
```

Comment:
Is `:tags` currently the only option? Maybe pass only the tags then (although I'm ok with this as well)?